src/EditorFeatures/Test/Preview/PreviewWorkspaceTests.cs

```diff
@@ -144,11 +144,13 @@ public void TestPreviewDiagnostic()
             diagnosticService.DiagnosticsUpdated += (s, a) => taskSource.TrySetResult(a);
 
             using var previewWorkspace = new PreviewWorkspace(VisualStudioMefHostServices.Create(EditorServicesUtil.ExportProvider));
+
             var solution = previewWorkspace.CurrentSolution
-.AddProject("project", "project.dll", LanguageNames.CSharp)
-.AddDocument("document", "class { }")
-.Project
-.Solution;
+                .WithAnalyzerReferences(new[] { DiagnosticExtensions.GetCompilerDiagnosticAnalyzerReference(LanguageNames.CSharp) })
```

Comment:
Seeing this at so many places, I am wondering if TestWorkspace.Create helpers should always add the compiler diagnostic analyzer as a solution analyzer reference. If there are tests that need to work without compiler analyzer, then probably we can add an option to `TestParameters` for it and default it to always have the compiler analyzer, which should be common.

Reply:
I don't think we should do that. There are plenty of tests that do not need these diagnostics. It's better if feature tests explicitly set up things they need. Tests testing a specific feature usually have a helper that sets the workspace up for the feature.